src/App.js

```diff
@@ -191,25 +190,21 @@ class App extends Component {
         <section id="main" className="container">
           {this.state.articles
             .filter(article => {
-              if (this.state.selectedSource === "All") {
-                return article;
-              } else if (article.source === this.state.selectedSource) {
-                return article;
-              }
-              return null;
+              return this.state.selectedSource === "All"
+                || article.source === this.state.selectedSource
             })
             .filter(article => {
```

Comment:
A small performance improvement would be to combine these two filters into a single filter - since it means your code would only check every item in the array once instead of (up to) twice before deciding to filter it out.